src/main/java/com/github/cameltooling/model/diagnostic/UnknownComponentErrorMsg.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cameltooling.model.diagnostic;
+
+import org.apache.camel.catalog.EndpointValidationResult;
+
+public class UnknownComponentErrorMsg implements CamelDiagnosticEndpointMessage<String> {
+    @Override
+    public String getErrorMessage(EndpointValidationResult result, String property) {
+        return "Unknown component / Missing dependency";
```

Comment:
is it possible to be more precise on what is unknown or missing? I guess it is the value of "property" that can be used?
If it is an unknown component why is there a "missing dependency"? Would worth explaining the link between an unknown component and a missing dependency to user.